src/framework/auth/services/token/token.ts

```diff
@@ -251,3 +253,48 @@ export class NbAuthOAuth2Token extends NbAuthSimpleToken {
     return JSON.stringify(this.token);
   }
 }
+
+/**
+ * Wrapper for OAuth2 token
+ */
+export class NbAuthOAuth2JWTToken extends NbAuthOAuth2Token {
+
+  static NAME = 'nb:auth:oauth2:jwt:token';
+
+  /**
+   * Returns access token payload
+   * @returns any
+   */
+  getAccessTokenPayload(): any {
+    return decodeJwtPayload(this.getValue())
+  }
+
+  /**
+   * for Oauth2 JWT token, the iat (issued at) field of the access_token payload
+   */
+  protected prepareCreatedAt(date: Date) {
+    let decoded;
+    try {
+       decoded = this.getAccessTokenPayload();
```

Comment:
so in case when OAuth2 token is JWT, the `iat` fields is a part of JWT access token, right?

Reply:
Yes exactly. Redundant with expires_in of the 'containing' token